cli/cmd/create-workspace.go

```diff
@@ -0,0 +1,149 @@
+
+	ws, err := c.CreateWorkspace(client, teamId, wsName)
+	if err != nil {
+		return fmt.Errorf("failed to create workspace: %w", err)
+	}
+
+	giturl := ""
+	if ws.GitUrl.Get() != nil {
+		giturl = *ws.GitUrl.Get()
+	}
+	branch := ""
+	if ws.InitialBranch.Get() != nil {
+		branch = *ws.InitialBranch.Get()
+	}
+	fmt.Println("Workspace created:")
+	fmt.Printf("\nID: %d\n", ws.Id)
+	fmt.Printf("Name: %s\n", ws.Name)
+	fmt.Printf("Team ID: %d\n", ws.TeamId)
+	fmt.Printf("Git Repository: %s\n", giturl)
+	fmt.Printf("Branch: %s\n", branch)
+	fmt.Printf("To open it in the Codesphere IDE run '%s open workspace -w %d'", os.Args[0], ws.Id)
+
+	return nil
+}
+
+func AddCreateWorkspaceCmd(create *cobra.Command, opts GlobalOptions) {
+	workspace := CreateWorkspaceCmd{
+		cmd: &cobra.Command{
+			Use:   "workspace",
+			Short: "Create a workspace",
+			Args:  cobra.RangeArgs(1, 1),
+			Long: out.Long(`Create a workspace in Codesphere.
+
+				Specify a (private) git repository or start an empty workspace.
+				Environment variables can be set to initialize the workspace with a specific environment.
+				The command will wait for the workspace to become running or a timeout is reached.
+
+				To decide which plan suits your needs, run 'cs list plans'
+			`),
+			Example: out.FormatExampleCommands("create workspace my-workspace", []out.Example{
+				{Cmd: "-p 20", Desc: "Create an empty workspace, using plan 20"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git", Desc: "Create a workspace from a git repository"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git -e FOO=BAR -e A=B", Desc: "Create a workspace and set environment variables"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git --vpn myVpn", Desc: "Create a workspace and connect to VPN myVpn"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git --timeout 30", Desc: "Create a workspace and wait 30 seconds for it to become running"},
+				{Cmd: "-r https://github.com/codesphere-cloud/landingpage-temp.git -b staging", Desc: "Create a workspace from branch 'staging'"},
+				{Cmd: "-r https://github.com/my-org/my-private-project.git -P", Desc: "Create a workspace from a private git repository"},
+			}),
+		},
+		Opts: CreateWorkspaceOpts{GlobalOptions: opts},
+	}
+	workspace.Opts.Repo = workspace.cmd.Flags().StringP("repository", "r", "", "Git repository to create the workspace from")
+	workspace.Opts.Vpn = workspace.cmd.Flags().String("vpn", "", "Vpn config to use")
+	workspace.Opts.Env = workspace.cmd.Flags().StringArrayP("env", "e", []string{}, "Environment variables to set in the workspace")
+	workspace.Opts.Plan = workspace.cmd.Flags().IntP("plan", "p", 8, "Plan ID for the workspace")
+	workspace.Opts.Private = workspace.cmd.Flags().BoolP("private", "P", false, "Use private repository")
+	workspace.Opts.Timeout = workspace.cmd.Flags().Duration("timeout", 60*time.Second, "Time to wait for the workspace to start")
```

Comment:
maybe add an example of the required format for the duration

Reply:
are we sure that 1 Minute is sensitive default? Maybe rather 10 minutes?

```suggestion
	workspace.Opts.Timeout = workspace.cmd.Flags().Duration("timeout", 1*time.Minute, "Time to wait for the workspace to start")
```